core/src/main/java/org/apache/iceberg/BaseOverwriteFiles.java

```diff
@@ -133,19 +133,19 @@ protected void validate(TableMetadata base, Snapshot snapshot) {
     }
 
     if (validateNewDataFiles) {
-      validateAddedDataFiles(base, startingSnapshotId, dataConflictDetectionFilter());
+      validateAddedDataFiles(base, startingSnapshotId, dataConflictDetectionFilter(), snapshot);
```

Comment:
Can we name this `parent` instead of `snapshot`? Snapshot is not very clear because there are a lot of snapshots.